test/test_core.jl

```diff
@@ -205,6 +205,26 @@ end
     end
 end
 
+@testset "equality & hashing" begin
+    # singletons (identity and property lens) are object equal
+    for (l, r) ∈ [
+        @lens(_) => @lens(_),
+        @lens(_.a) => @lens(_.a)
+    ]
+        @test l === r
+    end
+
+    # composite and index lenses are structurally equal
+    for (l, r) ∈ [
+        @lens(_[1]) => @lens(_[1])
+        @lens(_.a[2]) => @lens(_.a[2])
+    ]
+        @test l == r
+        @test hash(l) == hash(r)
+    end
+end
```

Comment:
Would be nice to also add tests that distinct stuff hashed/compares unequal: `@test @lens(_[1]) != @lens(_[2])`

Reply:
I added some more, check it out.